### PR TITLE
chore: split @Entity -> @EntityKey and @EntityType

### PR DESCRIPTION
**Pull request by octonato**
Relates to https://github.com/lightbend/kalix/issues/7614

This will also open the path to support generate ids. We don't have it yet, but to support it, we must be able to have methods without an id in the path so the proxy can generate a UUID on the fly.

### CODE REVIEW

File: samples/spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartEntity.java [hunk 1]
```diff
@@ -6,14 +6,16 @@
...
+import kalix.springsdk.annotations.EntityType;
 import kalix.springsdk.annotations.EventHandler;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.Collections;
 
 // tag::class[]
-@Entity(entityKey = "cartId", entityType = "shopping-cart") // <2>
-@RequestMapping("/cart/{cartId}") // <3>
+@Entity(entityKey = "cartId") // <2>
```
Comment (franciscolopezsancho) [inline samples/spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartEntity.java:16]: EntityKey instead?

File: docs/src/modules/spring/pages/event-sourced-entities.adoc [hunk 2]
```diff
@@ -60,11 +60,13 @@ Let's have a look at what our shopping cart entity will look like for the first
 include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartEntity.java[tag=class]
 ----
 <1> Create a class that extends `EventSourcedEntity<S>`, where `S` is the state type this entity will store (i.e. `ShoppingCart`).
-<2> Make sure to annotate such class with `@Entity` and pass in an `entityKey` and an `entityType`.
-<3> Make use of Spring's RequestMapping annotation to define the routes to your entity.
+<2> Make sure to annotate such class with `@EntityKey`  and pass name of the path param that will be used as the
```
Comment (franciscolopezsancho) [inline docs/src/modules/spring/pages/event-sourced-entities.adoc:63]:
```suggestion
<2> Annotate such class with `@EntityKey`  and pass the name of the key that will be used as the
```

Comment (franciscolopezsancho) [reply docs/src/modules/spring/pages/event-sourced-entities.adoc:63]: I would change other occurrences of `Make sure` to the use of the plain verb. Of course, just a suggestion.

File: sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/EntityType.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package kalix.springsdk.annotations;
+
+import java.lang.annotation.*;
+
+/**
+ * Assign a type to the entity. This name should be unique among the different existing entities
+ * within a Kalix application. Additionally, the name should be stable as a different name means a
+ * different entity in storage.
```
Comment (franciscolopezsancho) [inline sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/EntityType.java:24]: Would it be worth explaining why? What would happen if the name is not stable?

File: sdk/spring-sdk/src/test/java/kalix/springsdk/badwiring/eventsourced/IllDefinedEventSourcedEntity.java [hunk 4]
```diff
@@ -18,8 +18,10 @@
 
 import kalix.javasdk.eventsourcedentity.EventSourcedEntity;
 import kalix.springsdk.annotations.Entity;
+import kalix.springsdk.annotations.EntityType;
 import org.springframework.stereotype.Component;
 
-@Entity(entityKey = "id", entityType = "test")
+@Entity("id")
```
Comment (franciscolopezsancho) [inline sdk/spring-sdk/src/test/java/kalix/springsdk/badwiring/eventsourced/IllDefinedEventSourcedEntity.java:24]: EntityKey?

File: sdk/spring-sdk/src/test/java/kalix/springsdk/badwiring/valueentity/IllDefinedValueEntity.java [hunk 5]
```diff
@@ -18,8 +18,10 @@
 
 import kalix.javasdk.valueentity.ValueEntity;
 import kalix.springsdk.annotations.Entity;
+import kalix.springsdk.annotations.EntityType;
 import org.springframework.stereotype.Component;
 
-@Entity(entityKey = "id", entityType = "test")
+@Entity("id")
```
Comment (franciscolopezsancho) [inline sdk/spring-sdk/src/test/java/kalix/springsdk/badwiring/valueentity/IllDefinedValueEntity.java:24]: EntityKey?

File: samples/spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartEntity.java (hunk 1 above)
Comment (octonato) [reply samples/spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartEntity.java:16]: Oh! Missed this one.

File: sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/EntityType.java (hunk 3 above)
Comment (octonato) [reply sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/EntityType.java:24]: It says that a different name is a different entity in storage. Do you mean that we should be more explicit? 

I will try to find a better text. Stay tuned...

File: sdk/spring-sdk/src/test/java/kalix/springsdk/badwiring/valueentity/IllDefinedValueEntity.java (hunk 5 above)
Comment (octonato) [reply sdk/spring-sdk/src/test/java/kalix/springsdk/badwiring/valueentity/IllDefinedValueEntity.java:24]: Strange that not all of them were refactored. Those in the integration tests should have been included when I refactored using IntelliJ. 

Others are not visible to IntelliJ, like those in the samples.

Anyway, I will review them all.

Comment (octonato) [reply sdk/spring-sdk/src/test/java/kalix/springsdk/badwiring/valueentity/IllDefinedValueEntity.java]: I think I forgot to run the refactoring in IntelliJ. That's why all this classes were still using the old name.

**Review by octonato — COMMENTED** [review]
Fixed the previous issues and applied some suggestions. Ready for a new review round.

File: sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/EntityType.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+package kalix.springsdk.annotations;
+
+import java.lang.annotation.*;
+
+/**
+ * Assign a type to the entity. This name should be unique among the different existing entities
+ * within a Kalix application.
+ *
+ * Additionally, the name should be stable as a different name means a different entity in storage. Changing this name
+ * will create a new class of entity and all previous instances using the old name won't be accessible anymore.
```
Comment (octonato) [inline sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/EntityType.java:26]: @franciscolopezsancho, WDYT?

**Review by franciscolopezsancho — APPROVED** [review]
I think this change it's great.
LGTM

**Pull request merged**